scripts/west_commands/runners/blflash.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2021 Gerson Fernando Budke <nandojve@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Bouffalo Lab flash tool (blflash) runner for serial boot ROM'''
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+DEFAULT_BLFLASH_PORT = '/dev/ttyUSB0'
```

Comment:
not a great default

Reply:
I see you point here. However, all cheap boards are using an FTDI like serial converters and those enumerate as /dev/ttyUSB0. Only the development kit have a Cypress USB controller and it enumerates as /dev/ttyACM0.

@VynDragon ,

What is the default port enumerator on the bootloader?

Reply:
A few other reasons for `/dev/ttyUSB0` not working:
- Another FTDI-based devboard is currently plugged holding `/dev/ttyUSB0`
- Windows users might have `COM#` (with `#` some number), something different for MacOS too

Requiring the user to specify the port could help raising awareness that there might be some verification to do on their own, i.e. "is this an FTDI?", or "is this the right board?", or "is this correctly mapped from Windows through WSL2 or Linux VM?

Not sure which reason __@__**gmarull** had, in hope this helps.

Reply:
The blflash was replace by bflb_mcu_tool later. This is not an issue.

Reply:
> What is the default port enumerator on the bootloader?

ACM0 (or COM something on windows) but it's not supported by blflash, and only relevant for bl70x and bl61x which are currently not present here.
The only better option is see is trying to connect to all present serial ports like mpremote does, except that's probably dangerous for a flashing tool (and results in other stuffs like trying to flash bl61x with a bl60x firmware, happened to me before)